### PR TITLE
ocf_ssh: install system-wide shell environments

### PR DESCRIPTION
**Pull request by matthew-mcallister**
Once this is agreed upon, the next PR will unmount `/opt/ocf` and remove references to it.

### CODE REVIEW

File: modules/ocf_ssh/manifests/environment.pp [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  file {
+    '/etc/bash.bashrc':
+      source => 'puppet:///modules/ocf_ssh/environment/bash.bashrc';
+    '/etc/csh.cshrc':
+      source => 'puppet:///modules/ocf_ssh/environment/csh.cshrc';
+  }
+
+  exec { 'cat /home/s/st/staff/motd/motd > /etc/motd':
+    onlyif => 'test -f /home/s/st/staff/motd/motd'
+  }
```
Comment (chriskuehl) [inline modules/ocf_ssh/manifests/environment.pp]: can you just symlink?

**Comment by chriskuehl** [comment]
could we do this on all servers and desktops (so that any server you log in to uses a sane bashrc)?

i'd suggest making `ocf::packages::shells` which does `package { ['zsh', 'csh', 'tcsh', 'bash']:; }` and then adds the proper rc files in the right spot

if not, we should at least do it on apphost as well

File: modules/ocf_ssh/manifests/environment.pp (hunk 1 above)
Comment (matthew-mcallister) [inline modules/ocf_ssh/manifests/environment.pp]: Oh duh, NFS wasn't mounted on dev-tsunami at the time of writing.

**Comment by matthew-mcallister** [comment]
I agree and was going to suggest this, since I removed the references to /opt/ocf in my .bashrc and I'm not getting path set on supernova now.

**Comment by matthew-mcallister** [comment]
Replaced old diff.

File: modules/ocf/manifests/packages/shell.pp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class ocf::packages::shell {
+  package { ['bash', 'csh', 'tcsh', 'zsh']: }
```
Comment (chriskuehl) [inline modules/ocf/manifests/packages/shell.pp]: oops, probably don't actually want csh, my bad. tcsh will provide /bin/csh

Comment (matthew-mcallister) [inline modules/ocf/manifests/packages/shell.pp]: I was wondering why csh was missing from the list of packages.

File: modules/ocf/manifests/packages/shell.pp [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class ocf::packages::shell {
+  package { ['bash', 'tcsh', 'zsh']: }
+
+  file {
+    '/etc/bash.bashrc':
+      source => 'puppet:///modules/ocf/shell/bash.bashrc';
```
Comment (chriskuehl) [inline modules/ocf/manifests/packages/shell.pp]: require => Package['bash']

File: modules/ocf/manifests/packages/shell.pp [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    '/etc/csh.cshrc':
+      source => 'puppet:///modules/ocf/shell/csh.cshrc';
```
Comment (chriskuehl) [inline modules/ocf/manifests/packages/shell.pp]: require => Package['tcsh']

File: modules/ocf_ssh/manifests/environment.pp [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class ocf_ssh::environment {
+  file { '/etc/motd':
+    ensure => link,
+    target => '/home/s/st/staff/motd',
+  }
+}
```
Comment (chriskuehl) [inline modules/ocf_ssh/manifests/environment.pp]: suggestion: move to `ocf::motd` and `include ocf::motd` inside `ocf_ssh` and `ocf_apphost`

**Comment by chriskuehl** [comment]
otherwise shipit

File: modules/ocf/manifests/packages/shell.pp (hunk 3 above)
Comment (matthew-mcallister) [inline modules/ocf/manifests/packages/shell.pp]: I suppose, but is there a reason why?

File: modules/ocf_ssh/manifests/environment.pp (hunk 5 above)
Comment (matthew-mcallister) [inline modules/ocf_ssh/manifests/environment.pp]: Good idea.

File: modules/ocf/manifests/packages/shell.pp (hunk 3 above)
Comment (chriskuehl) [inline modules/ocf/manifests/packages/shell.pp]: during initial install, your file might be put in place before bash installs, then bash will install and want to replace that file (i think it probably won't for noninteractive installs though, so it's probably nbd)

File: modules/ocf_ssh/manifests/environment.pp (hunk 5 above)
Comment (chriskuehl) [inline modules/ocf_ssh/manifests/environment.pp]: maybe also on ocf_admin if you're bored

File: modules/ocf/manifests/packages/shell.pp (hunk 3 above)
Comment (matthew-mcallister) [inline modules/ocf/manifests/packages/shell.pp]: Oh, I didn't realize bash places something there by default. (Maybe not now, but it could in the future)

Comment (chriskuehl) [inline modules/ocf/manifests/packages/shell.pp]: it does:

```bash
ckuehl@supernova:~/ocf-proj$ dpkg -L bash | grep '^/etc'
/etc
/etc/bash.bashrc
/etc/skel
/etc/skel/.bash_logout
/etc/skel/.profile
/etc/skel/.bashrc
```

**Comment by matthew-mcallister** [comment]
I also removed remaining references to `/opt/ocf` since they were only in like 2 places.

File: modules/ocf_ldap/files/slapd.conf [hunk 6]
```diff
@@ -57,7 +57,7 @@ access to dn.subtree="ou=People,dc=OCF,dc=Berkeley,dc=EDU" attrs=userPassword
     by * none
 
 # prevent changing own shell when sorried
-access to dn.subtree="ou=People,dc=OCF,dc=Berkeley,dc=EDU" filter=(loginShell=/opt/ocf/bin/sorry) attrs=loginShell
+access to dn.subtree="ou=People,dc=OCF,dc=Berkeley,dc=EDU" filter=(loginShell=/opt/share/utils/sbin/sorry) attrs=loginShell
```
Comment (chriskuehl) [inline modules/ocf_ldap/files/slapd.conf]: utils/bin

Comment (matthew-mcallister) [inline modules/ocf_ldap/files/slapd.conf]: Actually should be utils/bin/sorried. Fixed in new diff.

**Pull request merged**